tests/examples/token/test_vipercoin.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+
+from ethereum import utils
+from ethereum.abi import ValueOutOfBounds
+from ethereum.tools import tester
+
+
+TOKEN_NAME = "Vipercoin"
+TOKEN_SYMBOL = "FANG"
+TOKEN_DECIMALS = 18
+TOKEN_INITIAL_SUPPLY = (21 * 10 ** 6)
+TOKEN_TOTAL_SUPPLY = TOKEN_INITIAL_SUPPLY * (10 ** TOKEN_DECIMALS)
+
+
+@pytest.fixture
+def token_tester():
+    t = tester
+    tester.s = t.Chain()
+    from viper import compiler
+    t.languages['viper'] = compiler.Compiler()
+    contract_code = open('examples/token/vipercoin.v.py').read()
+    tester.c = tester.s.contract(
+        contract_code,
+        language='viper',
+        args=[TOKEN_NAME, TOKEN_SYMBOL, TOKEN_DECIMALS, TOKEN_INITIAL_SUPPLY]
+    )
+    return tester
+
+
+@pytest.fixture
+def assert_tx_failed():
+    def assert_tx_failed(tester, function_to_test, exception=tester.TransactionFailed):
+        initial_state = tester.s.snapshot()
+        with pytest.raises(exception):
+            function_to_test()
+        tester.s.revert(initial_state)
+    return assert_tx_failed
+
+
+def test_initial_state(token_tester):
+    assert token_tester.c.totalSupply() == TOKEN_TOTAL_SUPPLY == token_tester.c.balanceOf(token_tester.accounts[0])
+    assert token_tester.c.balanceOf(token_tester.accounts[1]) == 0
+
```

Comment:
I'd test for all inputs (symbol, name, etc).  It's trivial but I think it provides a bit of clarity.

Reply:
:+1: